### ISSUE TITLE
Default Wayland logo in title bar for KDE

### ISSUE THREAD

**Issue by Avenred** [Vencord/Vesktop#376]
**Describe the bug**

The default Wayland logo appears in the title bar and in the task switcher when running Vesktop in KDE. The normal Vesktop logo appears in the taskbar as usual.

**To Reproduces**

1. Install Vesktop from Flatpak (not sure if this issues exists with other install methods)
1. Launch Vesktop on KDE
1. Observe the Wayland icon in the title bar
1. Press ALT + TAB and observe the same Wayland logo in the task switcher

**Expected behavior**

The Vesktop logo appears instead of the Wayland one.

**Screenshots**
![2024-02-03_17-39](https://github.com/Vencord/Vesktop/assets/13921970/954f26eb-37fa-4990-911c-e244ee66efa1)

**Desktop (please complete the following information):**
 - Fedora 39
 - KDE Plasma 5.27.10

**Command line output**

Run vesktop from the command line. Include the relevant command line output here:
```
Passing the following arguments to Electron: --ozone-platform-hint=auto
[3:0203/174127.763950:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0203/174128.049152:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0203/174128.049223:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
```

**Comment by Vendicated** [comment]
kde issue. works fine on gnome

try normal non flatpak version

**Comment by donut313** [comment]
this can be fixed by adding a window rule that points to
 the desktop file but it should just be fixed by default so it is a bug.
![2024_02_03_23:00:53](https://github.com/Vencord/Vesktop/assets/108238272/12cbd6f2-cde3-48b3-8ed3-6abfc549a9b6)

**Comment by EinfxchFinn** [comment]
![image](https://github.com/Vencord/Vesktop/assets/95358720/779c3196-9a3c-48ba-98ab-7b75c4b08bd3)
I dont have this issue on KDE Wayland. Both with and without `--ozone-platform-hint=auto` so this seems to be a flatpak specific problem

**Comment by Avenred** [comment]
> try normal non flatpak version

Works fine when run as an Appimage. Seems to be Flatpak specific.

**Comment by Vendicated** [comment]
perhaps related to not copying all resolutions here? not sure why we only copy 16x16 and 512x512 anyway, we should just copy all resolutions

https://github.com/flathub/dev.vencord.Vesktop/blob/fdf5ad4fcaf090339b5206f79654d00f3173b234/dev.vencord.Vesktop.yml#L35-L36

cc @lewisakura

**Comment by lewisakura** [comment]
> perhaps related to not copying all resolutions here? not sure why we only copy 16x16 and 512x512 anyway, we should just copy all resolutions

flatpak autogenerates the rest, we were told by the devs not to copy all of them

**Comment by KraXen72** [comment]
after the latest discord breakage & subsequent fixing, i started experiencing this issue.

**Comment by ludg1e** [comment]
This is partially fixed. The Wayland icon still shows in the window switcher (Alt+Tab), but not in the taskbar/task manager.
OS: Fedora 40
DE: Plasma 6.1.2

**Comment by fusuyfusuy** [comment]
Have the same issue. The icon is correct in the menu, but its showing wayland icon in the overview and the panel. This fixed it: https://www.reddit.com/r/kde/comments/tzw738/comment/i424ces/

**Comment by Vendicated** [comment]
Is this still an issue?

**Comment by electricbrass** [comment]
This issue seems to have reappeared. I don't remember for sure but I don't think I had this issue til recently, but now on my Steam Deck, its showing the generic X11 icon. The X11 window class/Wayland app_id and the desktop filename need to match. This is how many (though not all, GNOME notably uses the StartupWMClass instead of the filename) desktop environments match an application to the correct desktop file. Since Flatpak mandates a particular naming scheme for the desktop file, ideally the window class would be changed to match it, and the desktop file would be given the new name for the non-Flatpak packages as well, since there's no particular reason I know of not to use that naming scheme. It's even recommended by Freedesktop.org: https://specifications.freedesktop.org/desktop-entry/latest/file-naming.html

**Comment by OneNot** [comment]
> This issue seems to have reappeared. I don't remember for sure but I don't think I had this issue til recently, but now on my Steam Deck, its showing the generic X11 icon. The X11 window class/Wayland app_id and the desktop filename need to match. This is how many (though not all, GNOME notably uses the StartupWMClass instead of the filename) desktop environments match an application to the correct desktop file. Since Flatpak mandates a particular naming scheme for the desktop file, ideally the window class would be changed to match it, and the desktop file would be given the new name for the non-Flatpak packages as well, since there's no particular reason I know of not to use that naming scheme. It's even recommended by Freedesktop.org: https://specifications.freedesktop.org/desktop-entry/latest/file-naming.html

Yep. Showing the default Wayland icon for me in KDE Task Switcher. Though the icon is fine everywhere else as far as I saw. (CachyOS, KDE Plasma, Wayland, Flatpak version)

Is there a workaround? The standard KDE Window Rule fix doesn't seem to work.
